ttnn/cpp/ttnn/operations/normalization/groupnorm/groupnorm.hpp

```diff
@@ -0,0 +1,92 @@
+        return std::forward_as_tuple(input_tensor);
+    }
+
+    static inline const std::array<ttnn::TensorSchema, 1> input_tensor_schemas() {
+        return {ttnn::TensorSchema{
+            2, 4, {ttnn::bfloat16}, {ttnn::TILE_LAYOUT, ttnn::ROW_MAJOR_LAYOUT}, true, false, false, false}};
+    }
+
+    static inline ttnn::Tensor execute_on_worker_thread(
+        const ttnn::Tensor& input_tensor,
+        const int num_groups,
+        const float epsilon,
+        const std::optional<ttnn::Tensor>& input_mask = std::nullopt,
+        const std::optional<ttnn::Tensor>& weight = std::nullopt,
+        const std::optional<ttnn::Tensor>& bias = std::nullopt,
+        const std::optional<MemoryConfig>& memory_config = std::nullopt,
+        const std::optional<ttnn::DataType> dtype = std::nullopt,
+        std::optional<CoreGrid> core_grid = std::nullopt,
+        std::optional<bool> inplace = std::nullopt,
+        std::optional<ttnn::Layout> output_layout = std::nullopt) {
+        if (input_tensor.get_layout() == Layout::TILE and inplace.has_value()) {
+            TT_FATAL(inplace == false, "Tile layour does not support inplace tensors");
+        }
+        if (output_layout.has_value() and inplace.has_value()) {
+            if (output_layout != input_tensor.get_layout()) {
+                TT_FATAL(inplace == false, "cannot inplace tensors when layout are different");
+            }
+        }
+        TT_FATAL(core_grid.has_value(), "Automatic determination of grid size not supported");
+
+        TT_FATAL(input_tensor.is_sharded(), "Only sharded input tensors supported");
+
+        TT_FATAL(
+            input_tensor.memory_config().memory_layout != TensorMemoryLayout::WIDTH_SHARDED,
+            "Input tensor cannot be width sharded");
+
+        TT_FATAL(input_tensor.get_shape().rank() == 4, "Input tensor must be rank 4");
+
+        TT_FATAL(
+            input_tensor.get_shape()[-1] % num_groups == 0, "Number of channels must be divisible by number of groups");
+
+        const auto& ts = input_tensor.get_shape();
+        TT_FATAL(
+            (ts[0] * ts[1] * ts[2]) % ttnn::types::TILE_SIZE == 0,
+            "Input tensor dim NHW must be divisible by tile size");
+
+        const auto output_dtype = dtype.value_or(input_tensor.get_dtype());
+
+        const std::optional<ttnn::Tensor>& gamma =
+            weight.has_value() ? std::optional<ttnn::Tensor>(ttnn::unsqueeze_to_4D(weight.value())) : std::nullopt;
+        const std::optional<ttnn::Tensor>& beta =
+            bias.has_value() ? std::optional<ttnn::Tensor>(ttnn::unsqueeze_to_4D(bias.value())) : std::nullopt;
+
+        const MemoryConfig& dram_memory_config = tt::tt_metal::MemoryConfig{
+            .memory_layout = tt::tt_metal::TensorMemoryLayout::INTERLEAVED,
+            .buffer_type = tt::tt_metal::BufferType::DRAM};
+        const MemoryConfig& output_mem_config = memory_config.value_or(dram_memory_config);
```

Comment:
we often use passed memory_config or input's memory config. would it work here too, for consistency? plus in many cases thats actually better

Reply:
I'll make the change in group norm PR, this is just copied code from normalization.hpp, which is already in ttnn.